libkbfs/root_metadata.go

```diff
@@ -843,23 +391,46 @@ func (md *RootMetadata) swapCachedBlockChanges() {
 	}
 }
 
-// TlfHandleExtensions returns a list of handle extensions associated with the TLf.
-func (md *RootMetadata) TlfHandleExtensions() (extensions []TlfHandleExtension) {
-	if md.ConflictInfo != nil {
-		extensions = append(extensions, *md.ConflictInfo)
-	}
-	if md.FinalizedInfo != nil {
-		extensions = append(extensions, *md.FinalizedInfo)
+// A ConstRootMetadata is a thin wrapper around a
+// *RootMetadata. Functions that take a ConstRootMetadata parameter
+// must not store the parameter or modify it. Therefore, code that
+// passes a ConstRootMetadata to a function can assume that it is
+// neither modified nor stored.
+type ConstRootMetadata struct {
+	*RootMetadata
+}
```

Comment:
Agreed that would be nice, but I think in lots of places, the code just manipulates fields directly, so this would be a pretty major change.  Maybe a TODO if you don't see an easy way?